include/nanobind/eigen/dense.h

```diff
@@ -38,15 +60,12 @@ using array_for_eigen_t = ndarray<
         shape<(size_t) T::RowsAtCompileTime,
               (size_t) T::ColsAtCompileTime>>,
     std::conditional_t<
-        T::InnerStrideAtCompileTime == Eigen::Dynamic,
-        any_contig,
+        requiresContigMemory<T>,
         std::conditional_t<
-            T::IsRowMajor || NumDimensions<T> == 1,
+            NumDimensions<T> == 1 || T::IsRowMajor,
             c_contig,
-            f_contig
-        >
-    >
->;
+            f_contig>,
+        any_contig>>;
```

Comment:
This part wasn't clear to me. Why `any_contig` if `requiresContigMemory==false`?